.clang-tidy

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2020 The Orbit Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+---
+Checks: >
+  -*,
+  abseil-*,
+  bugprone-*,
+  cert-*,
+  -cert-err58-cpp,
+  cppcoreguidelines-*,
+  -cppcoreguidelines-special-member-functions,
+  -cppcoreguidelines-avoid-magic-numbers,
+  -cppcoreguidelines-pro-type-vararg,
+  -cppcoreguidelines-non-private-member-variables-in-classes,
+  -cppcoreguidelines-pro-type-reinterpret-cast,
```

Comment:
I would remove this. `reinterpret_cast` should really be warned about, even though we use it in some places. We can check theses cases and add a `// NOLINT` comment if it's really needed.

Reply:
I am unsure if you marked these 4 lines explicitly, or if you are just talking about the reinterpret_cast. (For the later I am okay with it and activated that check. 

For the other ones I am of the opinion that we should not activate them yet, or at all.
* "Magic numbers" are often used in tests as test data.
* The vararg check is triggered by the LOG and ERROR macros, which displays a warning everytime we use them (at least we should change those macros first)
* non-private member check: do we actually want to do that?

Reply:
Was totally by accident. I was only talking about reinterpret_cast.